lib/compiler.js

```diff
@@ -149,8 +149,8 @@ function findScripts(dir) {
 
   var results = [];
   files.forEach(function(filename) {
-    // ignore index.js and files prefixed with underscore
-    if ((filename === 'index.js') || (filename[0] === '_')) {
+    // ignore files prefixed with underscore
```

Comment:
I think you may have a flaw in your reasoning. In the example you described, `findScripts` is called on `boot` only, the entry `./custom-boot/scripts` from `bootScripts` is passed directly to `resolveRelativePaths` as far as I can tell.